src/client.rs

```diff
@@ -98,56 +147,65 @@ fn request(url: String, quarantine: bool) -> Boxed<Result<Response<Body>, String
 								})
 								.unwrap_or_default()
 								.to_string(),
+							true,
 							quarantine,
 						)
-						.await
-					} else {
-						match response.headers().get(header::CONTENT_ENCODING) {
-							// Content not compressed.
-							None => Ok(response),
-
-							// Content gzipped.
-							Some(hdr) => {
-								// Since we requested gzipped content, we expect
-								// to get back gzipped content. If we get
-								// back anything else, that's a problem.
-								if hdr.ne("gzip") {
-									return Err("Reddit response was encoded with an unsupported compressor".to_string());
-								}
-
-								// The body must be something that implements
-								// std::io::Read, hence the conversion to
-								// bytes::buf::Buf and then transformation into a
-								// Reader.
-								let mut decompressed: Vec<u8>;
-								{
-									let mut aggregated_body = match body::aggregate(response.body_mut()).await {
-										Ok(b) => b.reader(),
-										Err(e) => return Err(e.to_string()),
-									};
-
-									let mut decoder = match gzip::Decoder::new(&mut aggregated_body) {
-										Ok(decoder) => decoder,
-										Err(e) => return Err(e.to_string()),
-									};
-
-									decompressed = Vec::<u8>::new();
-									match io::copy(&mut decoder, &mut decompressed) {
-										Ok(_) => {}
-										Err(e) => return Err(e.to_string()),
-									};
-								}
-
-								response.headers_mut().remove(header::CONTENT_ENCODING);
-								response.headers_mut().insert(header::CONTENT_LENGTH, decompressed.len().into());
-								*(response.body_mut()) = Body::from(decompressed);
-
-								Ok(response)
+						.await;
+					};
+
+					match response.headers().get(header::CONTENT_ENCODING) {
+						// Content not compressed.
+						None => Ok(response),
+
+						// Content encoded (hopefully with gzip).
+						Some(hdr) => {
+							match hdr.to_str() {
+								Ok(val) => match val {
+									"gzip" => {}
+									"identity" => return Ok(response),
+									_ => return Err("Reddit response was encoded with an unsupported compressor".to_string()),
+								},
+								Err(_) => return Err("Reddit response was invalid".to_string()),
+							}
+
+							// We get here if the body is gzip-compressed.
+
+							// The body must be something that implements
+							// std::io::Read, hence the conversion to
+							// bytes::buf::Buf and then transformation into a
+							// Reader.
+							let mut decompressed: Vec<u8>;
+							{
+								let mut aggregated_body = match body::aggregate(response.body_mut()).await {
+									Ok(b) => b.reader(),
+									Err(e) => return Err(e.to_string()),
+								};
+
+								let mut decoder = match gzip::Decoder::new(&mut aggregated_body) {
+									Ok(decoder) => decoder,
+									Err(e) => return Err(e.to_string()),
+								};
+
+								decompressed = Vec::<u8>::new();
```

Comment:
Curious why this isn't initialized until here - declaring a new Vec doesn't allocate anything until you push onto it so it's okay

Reply:
Interesting, TIL. still, since it's harmless, I vote for keeping it as-is here.